tests/checkpoint/test_checkpoint_id_pk.py

```diff
@@ -0,0 +1,557 @@
+    What does this test?
+        - unexpected_index_column defined in Checkpoint only.
+        - COMPLETE output, which means we have `unexpected_index_list` and
+          `partial_unexpected_index_list`
+        - 1 Expectations added to suite
+    """
+    result = _build_checkpoint_and_run(
+        context=data_context_with_connection_to_metrics_db,
+        expectations=[expect_column_values_to_be_in_set],
+        asset_name=ANIMAL_ASSET,
+        result_format={"result_format": "COMPLETE", "unexpected_index_column_names": ["pk_1"]},
+    )
+    evrs = list(result.run_results.values())
+
+    index_column_names = evrs[0]["results"][0]["result"]["unexpected_index_column_names"]
+    assert index_column_names == ["pk_1"]
+
+    first_result_full_list = evrs[0]["results"][0]["result"]["unexpected_index_list"]
+    assert first_result_full_list == expected_unexpected_indices_output
+
+    first_result_partial_list = evrs[0]["results"][0]["result"]["partial_unexpected_index_list"]
+    assert first_result_partial_list == expected_unexpected_indices_output
+
+    unexpected_index_query: str = evrs[0]["results"][0]["result"]["unexpected_index_query"]
+    assert unexpected_index_query == expected_sql_query_output
+
+
+@pytest.mark.filesystem
+def test_sql_result_format_in_checkpoint_pk_defined_one_expectation_complete_output_with_query(
+    data_context_with_connection_to_metrics_db: FileDataContext,
+    expect_column_values_to_be_in_set: gxe.ExpectColumnValuesToBeInSet,
+    expected_unexpected_indices_output: list[dict[str, str | int]],
+    expected_sql_query_output: str,
+):
+    """
+    What does this test?
+        - unexpected_index_column defined in Checkpoint only.
+        - COMPLETE output, which means we have `unexpected_index_list` and
+         `partial_unexpected_index_list`
+        - 1 Expectations added to suite
+        - return_unexpected_index_query flag set to True
+    """
+    result = _build_checkpoint_and_run(
+        context=data_context_with_connection_to_metrics_db,
+        expectations=[expect_column_values_to_be_in_set],
+        asset_name=ANIMAL_ASSET,
+        result_format={
+            "result_format": "COMPLETE",
+            "unexpected_index_column_names": ["pk_1"],
+            "return_unexpected_index_query": True,
+        },
+    )
+    evrs = list(result.run_results.values())
+
+    index_column_names = evrs[0]["results"][0]["result"]["unexpected_index_column_names"]
+    assert index_column_names == ["pk_1"]
+
+    first_result_full_list = evrs[0]["results"][0]["result"]["unexpected_index_list"]
+    assert first_result_full_list == expected_unexpected_indices_output
+
+    first_result_partial_list = evrs[0]["results"][0]["result"]["partial_unexpected_index_list"]
+    assert first_result_partial_list == expected_unexpected_indices_output
+
+    unexpected_index_query = evrs[0]["results"][0]["result"]["unexpected_index_query"]
+    assert unexpected_index_query == expected_sql_query_output
```

Comment:
Are the asserts here suppose to be identical to the previous test, `test_sql_result_format_in_checkpoint_pk_defined_one_expectation_complete_output`?